array_analyzer/utils/io_utils.py

```diff
@@ -0,0 +1,97 @@
+import cv2 as cv
+from datetime import datetime
+import glob
+import natsort
+import numpy as np
+import os
+import skimage.io as io
+from skimage.color import rgb2grey
+
+
+def read_to_grey(path_, wellimage_):
+    """
+    a generator that receives file path and returns the next rgb image as greyscale and its name
+
+    :param str path_: path to folder with all images
+    :param str wellimage_: name of the file with image of the well.
+    :return: next image as greyscale np.ndarray
+    """
+    image_path = os.path.join(path_, wellimage_)
+    im = io.imread(image_path)
+    im = rgb2grey(im)
+    return im, os.path.basename(image_path)
+
+
+def read_gray_im(im_path):
+    """
+    Read image from full path to file location.
+    :param str im_path: Path to image
+    :return np.array im: Grayscale image
+    """
+    try:
+        im = cv.imread(im_path, cv.IMREAD_GRAYSCALE | cv.IMREAD_ANYDEPTH)
+    except IOError as e:
+        raise("Can't read image", e)
+    if not isinstance(im, np.ndarray):
+        raise IOError('Invalid image path')
+    return im
+
+
+def get_image_paths(input_dir):
+    """
+    Searches input directory and its subdirectory for all images
+    and returns sorted list of paths.
+
+    :param str input_dir: Input directory, may contain images or subdirectories
+        with one image each
+    :return dict well_images: Well name key, path to found image value
+    """
+    extensions = ('*.png', '*.tif', '*.jpg')
+
+    image_names = []
+    for ext in extensions:
+        search_str = os.path.join(input_dir, '**', ext)
+        image_names.extend(glob.glob(search_str, recursive=True))
+
```

Comment:
searching the folder tree recursively returns too many images, and assigns wrong images to the wells.
<img width="1441" alt="image" src="https://user-images.githubusercontent.com/2934183/80677235-722f6b00-8a6d-11ea-9757-2bf7cf3acb13.png">

Reply:
Above screenshot was taken with following arguments: 
`-e
-i="/Volumes/GoogleDrive/My Drive/ELISAarrayReader/images_scienion/2020-04-08-14-48-49-COVID_concentrationtest_April8_images"
-o="~/Documents/images_local/2020-04-08-14-48-49-COVID_concentrationtest_April8_images"
-wf=array_fit`

Reply:
> When I tested it on the data paths I didn't get these issues, but I never write outputs in the same directory as the input so that's probably why I missed that. Thanks for not only catching it but also fixing it!

Somewhat related -- do we want to generally allow writing reports and outputs to the same directory as the raw data files?  It does cause problems when, for example, searching for excel metadata spreadsheets.